test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,350 @@
+// Copyright 2024 Antrea Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"text/template"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/ptr"
+
+	crdv1beta1 "antrea.io/antrea/pkg/apis/crd/v1beta1"
+	agentconfig "antrea.io/antrea/pkg/config/agent"
+	"antrea.io/antrea/test/e2e/utils"
+)
+
+/*
+1) Create the KIND cluster.
+2) Once the cluster is up , create a service for custom DNS.
+3) Configure and update Antrea configmap with above service's IP.
+
+5) Create agnHost 2 pods (deployment?).
+6) Get IP of one of the pods of agnHost.
+7) create and configure the custom CoreDNS configMap with the IP received above.
+8) Create a custom DNS pod and configure it to use above configMap with agnHost pod's IP.
+9) Create and apply antrea FQDN policy.
+10) Deploy antrea-toolbox.
+
+11) curl the FQDN from within toolbox.
+12) imitate caching the IP belonging to above FQDN resolution by keeping it in a variable.
+13) change the existing IP in dns configmap with the other agnHost pod's IP.
+14) wait for new IP to get updated in configMap and let the changes be reflected in dns pod.
+15) curl the FQDN again with old IP , simulating usage of cache -- and it must fail with no connectivity.
+*/
+
+const (
```

Comment:
as a rule, I think you define too many constants. If all the tests defined that many single-use global constants, there would be a lot of name collisions in the package.

You should think about the following points:
- do I even need a constant; for example it is not clear that there is any value in defining `eventualWaitTime`
- should a function-level constant be defined instead? `testFullyQualifiedDomainName` could be a good candidate here


It seems unnecessary to define `customDnsPort` / `customDnsTargetPort`. Instead you could define a single function-scoped `dnsPort` constant.

Reply:
Even i felt at one point that there were just too many constants but i thought that i should be using them for situations like :
1) A configurable value , example ```eventualWaitTime``` but , i think it should have been useful if it was used at multiple places , hence changing a constant would have reflected the change at multiple places instead of changing values inidvidually.
2) In many cases i felt that anything which looks like a configuration value or something which needs to be hard coded may be put up as a constant.

but i get our point and also the consideration of using function scoped constants instead of global ones to realise the same thing.

Reply:
Some of these constants already exist in a way in framework.go
For example, `eventualWaitTime` can be replaced by `defaultTimeout` (and there is `defaultInterval` as well to go with it). Some tests may prefer a more appropriate value (based on what they are testing for) and it's fine to have a custom value, but there is no real value in defining a constant at that point, except perhaps a local (function-scope) one, right before it is used.

There are still way too many unnecessary global constants by the way.

Reply:
Yes, refactored to remove many such constants.